alts/src/main/java/io/grpc/alts/CheckGcpEnvironment.java

```diff
@@ -52,6 +52,10 @@ public static void check(boolean enableUntrustedAlts) {
   private static synchronized boolean isOnGcp() {
     if (cachedResult == null) {
       cachedResult = isRunningOnGcp();
+      if (cachedResult) {
+        // If we are running on GCE, enable DNS resolver.
+        System.setProperty("io.grpc.internal.DnsNameResolverProvider.enable_jndi", "true");
```

Comment:
This is still racy. The problem is that DnsNameResolver can be initialized before this method is called, in which case it will see "false." Depending on class load order you get different results.

Reply:
You are right. I think we shall pass this as command line for now instead of in the code, until grpc-java set DnsNameResolver with SRV querying as default.